### PR TITLE
Simplify SafeRepr a bit

### PR DESCRIPTION
**Pull request by bluetech**
Remove some dead code and make it easier to type-check.

The PR is split to commits.

### CODE REVIEW

File: src/_pytest/_io/saferepr.py [hunk 1]
```diff
@@ -2,65 +2,59 @@
...
-        if len(s) > self.maxstring:
-            i = max(0, (self.maxstring - 3) // 2)
-            j = max(0, self.maxstring - 3 - i)
-            s = repr(x[:i] + x[len(x) - j :])
-            s = s[:i] + "..." + s[len(s) - j :]
-        return s
+    def repr(self, x):
+        try:
+            s = super().repr(x)
+        except Exception as exc:
```
Comment (nicoddemus) [inline src/_pytest/_io/saferepr.py]: This `except` block is [not covered](https://codecov.io/gh/pytest-dev/pytest/compare/499fda234997d65b40e24f6fe3c7341f524b7d97...b47e1679e02029b847a9932e68cdf5db207991b8/diff), can you please add a test for it? Thanks!

**Comment by nicoddemus** [comment]
Even though this is an internal change, I suggest adding a `5603.trivial.rst` changelog entry anyway, with something along this lines:

```rst
Simplified internal ``SafeRepr`` class and remove some dead code.
```

This is useful so if someone has a regression because of the change this can help pinpoint the problem. 👍

**Comment by bluetech** [comment]
Thanks for the review @nicoddemus - updated.

**Comment by bluetech** [comment]
I'm going to take the "better ask for forgiveness" approach and merge this given the approval from @nicoddemus - please let me know if that's not appropriate.

**Pull request merged**